source/server-discovery-and-monitoring/tests/unified/backoff-heartbeat-failure.yml

```diff
@@ -0,0 +1,94 @@
+description: heartbeat-failure-clears-backoff-pool
+schemaVersion: "1.28"
+runOnRequirements:
+  - minServerVersion: "4.4"
+    serverless: forbid
+    topologies:
+      - single
+      - replicaset
+      - sharded
+createEntities:
+  - client:
+      id: setupClient
+      useMultipleMongoses: false
+initialData:
+  - collectionName: heartbeat-backoff-error
+    databaseName: sdam-tests
+    documents:
+      - _id: 1
+      - _id: 2
+tests:
+  - description: A heartbeat failure during backoff should clear the pool
+    operations:
+      - name: createEntities
+        object: testRunner
+        arguments:
+          entities:
+            - client:
+                id: client
+                useMultipleMongoses: false
+                observeEvents:
+                  - commandStartedEvent
+                  - poolBackoffEvent
+                  - poolClearedEvent
+                  - serverHeartbeatFailedEvent
+                  - serverHeartbeatSucceededEvent
+                uriOptions:
+                  retryWrites: false
+                  heartbeatFrequencyMS: 10000
+                  serverMonitoringMode: poll
+                  appname: heartbeatBackoffFailTest
+            - database:
+                id: database
+                client: client
+                databaseName: sdam-tests
+            - collection:
+                id: collection
+                database: database
+                collectionName: heartbeat-backoff-error
+      - name: waitForEvent
+        object: testRunner
+        arguments:
+          client: client
+          event:
+            serverHeartbeatSucceededEvent: {}
+          count: 1
+      - name: failPoint
+        object: testRunner
+        arguments:
+          client: setupClient
+          failPoint:
+            configureFailPoint: failCommand
+            mode: alwaysOn
+            data:
+              failCommands:
+                - isMaster
+                - hello
+              appName: heartbeatBackoffFailTest
+              closeConnection: true
+      - name: insertMany
+        object: collection
+        arguments:
+          documents:
+            - _id: 3
+            - _id: 4
+        expectError:
+          isError: true
+      - name: waitForEvent
+        object: testRunner
+        arguments:
+          client: client
+          event:
+            serverHeartbeatFailedEvent: {}
+          count: 1
+    expectEvents:
+      - client: client
+        eventType: cmap
+        events:
+          - poolBackoffEvent: {}
+          - poolBackoffEvent: {}
+          - poolBackoffEvent: {}
+          - poolBackoffEvent: {}
+          - poolBackoffEvent: {}
+          - poolBackoffEvent: {}
```

Comment:
For this, and the tests below: where are all these pool backoff events coming from?  I get zero / one (depending on my implementation, see below) of them in Node.

This is related to my [comment in the design](https://docs.google.com/document/d/1teqNgeWbW6dpRQOALrJTEBRoO6sYcrpq9T3_NJE0QfU/edit?disco=AAABtWIuBsk).  Even if I align my implementation with the design, I only get one backoff event because there are no establishment attempts:

- there are no requests in the wait queue
- minPoolSize is 0, so no background thread population

Reply:
two follow ups to the new tests:

- can we clarify in the CMAP spec that drivers are supposed to add the SystemOverloadError to network errors during connection establishment?
- can we use the `errorLabelsContain` field in the unified test format to assert on this as well, in the tests you've added?  Something like:

```
      - name: insertMany
        object: collection
        arguments:
          documents:
            - _id: 3
            - _id: 4
        expectError:
          isError: true
          errorLabelsContain: 
              - 'SystemOverloadError'
```